src/parsers.rs

```diff
@@ -115,6 +136,103 @@ impl<'a> Parser<'a> {
         Err(())
     }
 
+    #[cfg(feature = "ipv6")]
+    fn accept_ipv6_part(&mut self, (head, tail): (&mut [u16; 8], &mut [u16; 6]),
```

Comment:
:+1: An excellent job refactoring this method, it is very easy to see now that it is correct.